src/library/scala/reflect/ClassTag.scala

```diff
@@ -69,18 +57,7 @@ trait ClassTag[T] extends ClassManifestDeprecatedApis[T] with Equals with Serial
    * is uncheckable, but we have an instance of `ClassTag[T]`.
    */
   def unapply(x: Any): Option[T] =
-    if (null != x && (
-            (runtimeClass.isInstance(x))
-         || (x.isInstanceOf[Byte]    && runtimeClass.isAssignableFrom(classOf[Byte]))
-         || (x.isInstanceOf[Short]   && runtimeClass.isAssignableFrom(classOf[Short]))
-         || (x.isInstanceOf[Char]    && runtimeClass.isAssignableFrom(classOf[Char]))
-         || (x.isInstanceOf[Int]     && runtimeClass.isAssignableFrom(classOf[Int]))
-         || (x.isInstanceOf[Long]    && runtimeClass.isAssignableFrom(classOf[Long]))
-         || (x.isInstanceOf[Float]   && runtimeClass.isAssignableFrom(classOf[Float]))
-         || (x.isInstanceOf[Double]  && runtimeClass.isAssignableFrom(classOf[Double]))
-         || (x.isInstanceOf[Boolean] && runtimeClass.isAssignableFrom(classOf[Boolean]))
-         || (x.isInstanceOf[Unit]    && runtimeClass.isAssignableFrom(classOf[Unit])))
-       ) Some(x.asInstanceOf[T])
+    if (null != x && (runtimeClass.isInstance(x))) Some(x.asInstanceOf[T])
```

Comment:
`null != x &&` is totally redundant now, because `isInstance(x)` always returns `false` when `x` is `null`.

Reply:
thanks, will fix in next commit